### PR TITLE
Fixes for RPM Spec: Move the .so to devel files

### PR DESCRIPTION
**Pull request by dgarske**
# Description

Move the .so to devel files. The pure *.so file is considered a dev file.

# Testing

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

File: rpm/spec.in [hunk 1]
```diff
@@ -37,6 +40,7 @@ you will need to install %{name}-devel.
 %setup -q
 
 %build
+autoreconf -fi
```
Comment (julek-wolfssl) [inline rpm/spec.in:43]: This is building from the dist tarball so this step is unnecessary.

File: rpm/spec.in [hunk 2]
```diff
@@ -13,6 +13,9 @@ BuildRequires: make
 BuildRequires: pkgconfig
 BuildRequires: sed
 BuildRequires: tar
+BuildRequires: autoconf
+BuildRequires: automake
+BuildRequires: libtool
```
Comment (julek-wolfssl) [inline rpm/spec.in:18]: This is building from the dist tarball so these are not build dependencies.

File: rpm/spec.in [hunk 3]
```diff
@@ -49,7 +53,7 @@ fi
 %install
 %{__rm} -rf %{buildroot}
 %{__make} install  DESTDIR="%{buildroot}" AM_INSTALL_PROGRAM_FLAGS=""
-mkdir -p $RPM_BUILD_ROOT/
+%{__install} -m 0644 wolfssl/options.h %{buildroot}/usr/include/wolfssl/options.h
```
Comment (julek-wolfssl) [inline rpm/spec.in:56]: Why is this needed now?

**Review by julek-wolfssl — CHANGES_REQUESTED** [review]

**Comment by dgarske** [comment]
Retest this please

**Review by SparkiDev — APPROVED** [review]

**Comment by SparkiDev** [comment]
retest this please

**Comment by dgarske** [comment]
Retest this please

File: rpm/spec.in (hunk 3 above)
Comment (sergey-safarov) [reply rpm/spec.in]:
```
mkdir -p $RPM_BUILD_ROOT/
```
Use less here and can be removed

Comment (dgarske) [reply rpm/spec.in]: Fixed https://github.com/wolfSSL/wolfssl/pull/7086
